### DISCUSSION TITLE
Heading of bodies rising/setting ?

### DISCUSSION THREAD

**Discussion by achalmers** [cosinekitty/astronomy#228] [Q&A] [answered]
Im using Astronomy_SearchRiseSet which is working fine but I'd like to know the heading of were the body is ... to help someone with boinoculars locate it.

**Comment by cosinekitty** [comment, marked as answer]
Hi @achalmers. The easiest way to do this is to use the time returned by Astronomy_SearchRiseSet, and call [Astronomy_Horizon](https://github.com/cosinekitty/astronomy/tree/master/source/c#Astronomy_Horizon) with the same body and time. The return value will be horizontal coordinates in the type [astro_horizon_t](https://github.com/cosinekitty/astronomy/tree/master/source/c#astro_horizon_t). The `azimuth` field will tell you what compass direction to look. The azimuth value is oriented such that 0=North, 90=East, 180=South, 270=West.

**Reply by achalmers** [reply]
Yes. Thx for the pointers.  I am now calculating rise and set times and heading for planets within a few mins/degrees of what TheSkyLive.com displays  . To help others, my code looks like :

    //Test for Boston MA 
    String test_location = "BOSTON MA";
    astro_time = Astronomy_MakeTime(2022, 06, 01, 0, 0, 0) ;                                            
    astro_observer_t observer = Astronomy_MakeObserver((double)42.0+(21/60.0)+30/3600, (double)-70.0-(3/60)-35/3600, 9.0);

      Serial.println("Venus");
      venusrise  = Astronomy_SearchRiseSet(BODY_VENUS,  observer, DIRECTION_RISE, astro_time, 300.0); //find time of next rise and set of VENUS
      venusset   = Astronomy_SearchRiseSet(BODY_VENUS,  observer, DIRECTION_SET,  astro_time, 300.0);

      eqd = Astronomy_Equator(BODY_VENUS, &venusrise.time, observer, EQUATOR_OF_DATE, ABERRATION);  //convert 
      hor = Astronomy_Horizon(&venusrise.time, observer, eqd.ra, eqd.dec, REFRACTION_NORMAL);

      
      status = Astronomy_FormatTime(venusrise.time, TIME_FORMAT_SECOND2, text, sizeof(text));   //made format suitable for Excel csv import
      Serial.print(", Venus rises , ");Serial.print(text);

      status = Astronomy_FormatTime(venusset.time, TIME_FORMAT_SECOND2, text, sizeof(text));
      Serial.print(" ,Venus sets , ");Serial.print(text);

      Serial.print(", AND VENUS rises at heading : , "); Serial.println(hor.azimuth);

**Comment by cosinekitty** [comment]
As a clarification, you will need to call Astronomy_Equator, asking for EQUATOR_OF_DATE coordinates and enabling aberration. This will get you the angular RA (right ascension) and DEC (declination) coordinates, that you then pass to Astronomy_Horizon. Something like this:

```c
astro_equatorial_t eqd;
astro_horizon_t hor;

eqd = Astronomy_Equator(body, &time, observer, EQUATOR_OF_DATE, ABERRATION);
hor = Astronomy_Horizon(&time, observer, eqd.ra, eqd.dec, REFRACTION_NORMAL);
printf("compass direction = %0.3lf\n", hor.azimuth);
```

**Comment by cosinekitty** [comment]
@achalmers Did the above answer your question sufficiently?

**Comment by achalmers** [comment]
Yes it did, thx.
I posted my code so others can quickly get the position and azimuth.
A
